control-plane-operator/controllers/hostedcontrolplane/pki/csisnapshotcontroller.go

```diff
@@ -0,0 +1,21 @@
+package pki
+
+import (
+	"fmt"
+
+	"github.com/openshift/hypershift/support/config"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Create TLS keys for csi-snapshot-webhook.
+// In standalone OCP it's created automatically when csi-snapshot-controller-operator creates Service for
+// the webhook with annotation `service.openshift.io/serving-cert-secret-name`, in HyperShift
+// it must be done by control-plane-operator.
```

Comment:
fwiw it doesn't really must be done. It's done as a best effort to preserve the ability to run in vanilla management clusters

Reply:
I don't get it. If control-plane-operator does not generate TLS config for a webhook that runs in the mgmt cluster, who else is going to generate it?